src/components/TradesWidget/TradeRow.tsx

```diff
@@ -121,40 +176,57 @@ export const TradeRow: React.FC<TradeRowProps> = (params) => {
     }
   }, [orderSellAmount, sellAmount, sellToken, sellTokenDecimals, type])
 
-  const market = useMemo(() => `${displayTokenSymbolOrLink(buyToken)}/${displayTokenSymbolOrLink(sellToken)}`, [
-    buyToken,
-    sellToken,
-  ])
+  const [isPriceInverted, setIsPriceInverted] = useState(false)
+
+  const {
+    side,
+    market,
+    quoteTokenLabel,
+    sellTokenLabel,
+    buyTokenLabel,
+    fillPrice: marketFillPrice,
+    limitPrice: marketLimitPrice,
+  } = useMemo(
+    () =>
+      getMarketAndPrices({
+        sellToken,
+        buyToken,
+        isPriceInverted,
+        limitPrice,
+        fillPrice,
+      }),
+    [sellToken, buyToken, isPriceInverted, limitPrice, fillPrice],
+  )
 
   // Do not display trades that are not settled
   return !isTradeSettled(trade) ? null : (
     <TradeRowFoldableWrapper data-order-id={orderId} data-batch-id={batchId}>
       <td data-label="Date" className="showResponsive" title={date.toLocaleString()}>
         {formatDistanceStrict(date, new Date(), { addSuffix: true })}
       </td>
-      <td
-        data-label="Market"
-        className="showResponsive"
-        onClick={(): void =>
-          onCellClick({
-            target: { value: market },
-          })
-        }
-      >
-        {market}
+      <td data-label="Market" className="showResponsive">
+        <SwapIcon swap={(): void => setIsPriceInverted(!isPriceInverted)} />{' '}
+        <span
+          onClick={(): void =>
+            onCellClick({
+              target: { value: market },
+            })
+          }
```

Comment:
This looks very artificial. I understand that the func expects a proper MouseEvent, but maybe it should be changed to accept a value.